rudder-web/src/main/scala/com/normation/rudder/web/snippet/administration/PropertiesManagement.scala

```diff
@@ -900,4 +904,84 @@ class PropertiesManagement extends DispatchSnippet with Loggable {
         } )
     } ) apply xml
   }
+
+  def nodeOnAcceptDefaultsConfiguration = { xml : NodeSeq =>
+
+    val modes = SelectableOption[Option[PolicyMode]](None, "Use global value") :: PolicyMode.allModes.map { x =>
+      SelectableOption[Option[PolicyMode]](Some(x), x.name.capitalize)
+    }.toList
+    // node states, sorted [init, enable, other]
+    val states = NodeState.values.toList.map { x => x match {
+      case NodeState.Initializing  => (0, x, x.name.capitalize)
+      case NodeState.Enabled       => (1, x, x.name.capitalize)
+      case NodeState.EmptyPolicies => (2, x, "Empty policies")
+      case NodeState.Ignored       => (3, x, x.name.capitalize)
+      case NodeState.PreparingEOL  => (4, x, "Preparing End Of Life")
+    } }.sortBy( _._1 ).map{ case (_, x, label) =>
+      SelectableOption[NodeState](x, label)
+    }
+
+    val process = (for {
+      initialPolicyMode <- configService.rudder_node_onaccept_default_policy_mode()
+      initialNodeState  <- configService.rudder_node_onaccept_default_state()
+    } yield {
+
+      // initialise values for the form (last time it was saved and current user input)
+      var initPolicyMode = initialPolicyMode
+      var initNodeState = initialNodeState
+
+      var policyMode = initialPolicyMode
+      var state = initialNodeState
+
+      def noModif() = initPolicyMode == policyMode && initNodeState == state
+      def check() = {
+        S.notice("nodeOnAcceptDefaultsMsg","")
+        Run(s"""$$("#nodeOnAcceptDefaultsSubmit").attr("disabled",${noModif()});""")
+      }
+
+      def submit() = {
+        val save =
+          for {
+            _ <- configService.set_rudder_node_onaccept_default_state(state)
+            _ <- configService.set_rudder_node_onaccept_default_policy_mode(policyMode)
+          } yield {
+            ()
+          }
+
+
+        S.notice("nodeOnAcceptDefaultsMsg", save match {
+          case Full(_)  =>
+            initNodeState = state
+            initPolicyMode = policyMode
+
+            "Node default configuration post-acceptation correctly saved. Next accepted node will get these default configuration."
+          case eb: EmptyBox =>
+            "There was an error when updating node default proerties"
+        } )
+        check()
+      }
+
+      "#nodeOnAcceptState" #> SHtml.ajaxSelectObj(states, Full(initNodeState), { (x:NodeState) => state = x; check}, ("id","nodeOnAcceptState")) &
+      "#nodeOnAcceptPolicyMode" #> SHtml.ajaxSelectObj(modes, Full(initPolicyMode), { (x: Option[PolicyMode]) => policyMode = x; check}, ("id","nodeOnAcceptPolicyMode")) &
+      "#nodeOnAcceptDefaultsSubmit" #> {
+        SHtml.ajaxSubmit("Save changes", submit _, ("class","btn btn-default"))
+      } &
+        "#nodeOnAcceptDefaultsSubmit *+" #> {
+        Script(check())
+      }
+
+    }) match {
+      case eb: EmptyBox =>
+        val fail = eb ?~ "there was an error while fetching value of property: 'directive script engine'"
```

Comment:
i'm not sure it's the directive engine - shouldn't it be the 'node on accept parameters'

Reply:
nice catch!